src/lib/encoder.ts

```diff
@@ -325,6 +326,10 @@ const valueTypeEncodingMap = {
     decode: (value: string) => abiCoder.decodeParameter('address', value),
   },
   // NOTE: We could add conditional handling of numeric values here...
+  uint128: {
+    encode: (value: string | number) => padLeft(value, 32),
+    decode: (value: string) => toDecimal(value),
+  },
```

Comment:
You can't do like this?

```suggestion
  uint128: {
    encode: (value: string | number) => abiCoder.encodeParameter('uint128', value),
    decode: (value: string) => abiCoder.decodeParameter('uint128', value),
  },
```

Reply:
No this does not work work because  it uses ABI encoding.

ABI encoding will always pas to 32 bytes, even if the type is smaller.

See the Solidity docs:

https://docs.soliditylang.org/en/v0.8.20/abi-spec.html

![image](https://github.com/ERC725Alliance/erc725.js/assets/31145285/47b9de4d-3e49-4c42-80bf-ac0a91848ed8)

Reply:
Ok I see, nice one, ty for the explanation